File: Apps/Codaxy.Dextop.Showcase/Demos/Windows/WindowOnDemand.js
Ext.ns('Showcase.demos');


Ext.define('Showcase.demos.WindowOnDemandWindow', {
	extend: 'Dextop.Window',

	autoWidth: true,
	autoHeight: true,
	title: 'Window On Demand',
	
	initComponent: function() {
		Ext.apply(this, {
			items: {
				xtype: 'button',
				text: 'Create Simple Window',
				margin: '5 5 5 5',
				scope: this,	
				handler: this.createSimpleWindow				
			}
		})
		
		this.callParent(arguments);
	},
	
	createSimpleWindow: function() {
		this.remote.CreateWindow('simple', {
			success: function(config) {
				var win = Dextop.create(config);
				win.show();
			}
		});
	}
});

File: Apps/Codaxy.Dextop.Showcase/Demos/Windows/WindowOnDemand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using Codaxy.Dextop.Remoting;

namespace Codaxy.Dextop.Showcase.Demos.Windows
{
    [Demo("WindowOnDemand",
        Title = "Window On Demand",
        Description = "Creating server-side window on demand",
        Path = "~/Demos/Windows"
    )]
    [LevelBeginner]
    [TopicDextopWindows]
    [CategoryCodeSnippet]
    public class WindowOnDemandWindow : DextopWindow
    {
        public override void InitRemotable(DextopRemote remote, DextopConfig config)
        {
            base.InitRemotable(remote, config);
            Remote.RemoteHostType = "Showcase.demos.WindowOnDemandWindow";
        }

        [DextopRemotable]
        public DextopConfig CreateWindow(String type)
        {
            IDextopRemotable window;
            switch (type)
            {
                case "simple":
                    window = new SimpleWindow();
                    break;
                // ...
                default:
                    throw new DextopErrorMessageException("Unknown window type: {0}", type);
            }

            return Remote.Register(window);
        }
    }
}
